pennylane/transforms/hamiltonian_expand.py

```diff
@@ -106,26 +122,34 @@ def hamiltonian_expand(tape, group=True):
             "Passed tape must end in `qml.expval(H)`, where H is of type `qml.Hamiltonian`"
         )
 
-    if group:
-        hamiltonian.simplify()
-        return qml.transforms.measurement_grouping(tape, hamiltonian.ops, hamiltonian.coeffs)
+    if group or hamiltonian.grouping_indices is not None:
+        # use groups of observables if available or explicitly requested
+        coeffs_groupings, obs_groupings = hamiltonian.get_groupings()
+    else:
+        # else make each observable its own group
+        obs_groupings = [[obs] for obs in hamiltonian.ops]
+        coeffs_groupings = [hamiltonian.coeffs[i] for i in range(len(hamiltonian.ops))]
```

Comment:
Oh, you mean I should use the ``terms`` here and iterate once?

Reply:
yeah, my first comment was along the lines of, you could replace the two list comprehensions with a single for loop over `hamiltonian.ops`.

In my second comment, I was then wondering if the for loop was needed at all, since we're not really computing any new information here. Rather than 'reshaping' the info so that the logic below can be shared, maybe it's better to just have different logical branches

Reply:
I removed the loop, you are right!